### ISSUE TITLE
TF 1.2 vs 1.1: Keras K.set_learning_phase(False) not working in 1.2 but works in 1.1

### ISSUE THREAD

**Issue by dylanrandle** [tensorflow/tensorflow#11336]
Hi all,

Works fine in 1.1 but in 1.2: 
```
from tensorflow.contrib.keras.python.keras import backend as K
from tensorflow.contrib.keras.python.keras.models import load_model
K.set_learning_phase(False)
model = load_model(MODEL_PATH)
```
```
model.uses_learning_phase <-- returns TRUE
```

This does not happen in 1.1. What may have changed that cause this in 1.2? 

Thank you in advance.
Best regards,
Dylan Randle

**Comment by cy89** [comment]
Please provide details about what platform you are using  (operating system, architecture). Also include your TensorFlow version. Also, did you compile from source or install a binary?  Make sure you also include the exact command if possible to produce  the output included in your test case. If you are unclear what to include see the issue template displayed in [the Github new issue template](https://github.com/tensorflow/tensorflow/issues/new).

We ask for this in the issue submission template, because it is really difficult to help without that information. Thanks!

**Comment by dylanrandle** [comment]
Hi, Sorry for that. 

Ubuntu 16.04 
Tensorflow version is 1.2 (install binary)

To clarify, the entire script is above, except:
`model.uses_learning_phase <-- returns TRUE` 
is really
```
if model.uses_learning_phase:
    raise RuntimeError('Model using learning phase.')
```

**Comment by cy89** [comment]
@fchollet could you comment on this? I don't understand some of the subtleties. It looks to me as if "uses_learning_phase" is intended to indicate whether a model behaves differently during learning or inference. But I think that the "set_learning_phase" flag is just indicating whether learning or inference is being done.

**Comment by dylanrandle** [comment]
Hey all,

Is there any update about this? Has anyone experienced the issue I am talking about? 

It's not crucial because I can use tensorflow 1.1 to accomplish to task, but it would be nice if it could be fixed? 

Thanks,
Dylan

**Comment by fchollet** [comment]
`model.uses_learning_phase` indicates whether the model is supposed to have a different behavior in inference and training. It's basically for regularization (dropout) and BatchNorm.

The "learning phase" is a flag is a flag which indicates training/inference. It is set to 1 when using e.g. `fit` and to 0 when using e.g. `predict`.

`K.set_learning_phase(False)` sets the "learning phase" to be always 0, i.e. `fit` will have the model behave in inference mode (e.g. no dropout and BatchNorm behavior set to inference).

The value returned by `model.uses_learning_phase` depends only on the model and is independent from `K.set_learning_phase(False)`.

**Comment by dylanrandle** [comment]
Hi @fchollet Thank you very much. That solves my issue, however I'm still concerned about the different behaviour?

Is there a reason why, for the exact same model, I would be getting model.uses_learning_phase <- False in Tf 1.1 and True in TF 1.2?

Btw, great article yesterday! Looking forward to today's! :D

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
A member of the TensorFlow organization has replied after the stat:awaiting tensorflower label was applied.

**Comment by drpngx** [comment]
Closing due to inactivity.
